delphi-frontend/src/main/antlr3/au/com/integradev/delphi/antlr/Delphi.g

```diff
@@ -456,9 +457,14 @@ enumType                     : '('<EnumTypeNodeImpl>^ (enumTypeElement (',')?)*
                              ;
 enumTypeElement              : nameDeclaration ('=' expression)? -> ^(TkEnumElement<EnumElementNodeImpl> nameDeclaration expression?)
                              ;
-typeReference                : stringType
+typeReference                : nameReference -> ^(TkTypeReference<TypeReferenceNodeImpl> nameReference)
+                             ;
+typeReferenceOrString        : stringType
+                             | typeReference
+                             ;
+typeReferenceOrStringOrFile  : stringType
                              | FILE<FileTypeNodeImpl>^
```

Comment:
Why is this specifically *just* the `file` keyword and not `fileType`?

Reply:
`file of x` can't be substituted in place of a name reference, it's only allowed as a type declaration.

Reply:
This probably applies to `string[x]` too, so we may need to adjust the `stringType` usage here.

Reply:
Can confirm that this applies to `string[x]` too, I've replaced these `stringType` usages with `STRING<StringTypeNodeImpl>^`.

Reply:
Gotcha.